src/types.rs

```diff
@@ -171,20 +182,21 @@ impl<T: Clone> IntoIterator for DVec<T> {
     type IntoIter = std::vec::IntoIter<Self::Item>;
 
     fn into_iter(self) -> Self::IntoIter {
+        #[allow(clippy::unnecessary_to_owned)]
         self.0.to_vec().into_iter()
     }
 }
 
-impl<'a, T> TryFrom<&'a Vec<T>> for DVec<T>
+impl<'a, T> TryFrom<&'a [T]> for DVec<T>
 where
     T: Clone,
 {
     type Error = &'static str;
 
-    fn try_from(value: &'a Vec<T>) -> Result<Self, Self::Error> {
+    fn try_from(value: &'a [T]) -> Result<Self, Self::Error> {
         match value.len() {
             2 => Ok(DVec([value[0].clone(), value[1].clone()])),
-            _ => Err("SizeError"),
+            _ => Err("SizeError: length 2 was expected"),
```

Comment:
maybe we can try to constraint `value` to be of size of 2 by another way?

Reply:
I don't really know how to do...